### ISSUE TITLE
Stack overflow causes segmentation fault with OCaml 4.12.0 & MacOS 11 (Apple M1)

### ISSUE THREAD

**Issue by IagoAbal** [ocaml/ocaml#10547]
We are experiencing segmentation faults as a result of stack overflows caused by some large inputs. Need to confirm this for Linux, but in MacOS 11 it is very easy to reproduce, see below.

segfault.ml
```ocaml
let rec stack_overflow x = 1 + stack_overflow (x+1)

let () = 
  stack_overflow 0 |> ignore
```

```bash
% ocamlopt --version
4.12.0
% ocamlopt segfault.ml -o segfault          
% ./segfault 
zsh: segmentation fault  ./segfault
```

**Comment by gadmm** [comment]
Stack overflow detection actually is not yet implemented on ARM+macos. It is probably just a matter of adapting https://github.com/ocaml/ocaml/blob/trunk/runtime/signals_osdep.h#L38-L66 to the arm64 target and defining `HAS_STACK_OVERFLOW`.

I wrote the following notes when I last dug into the implementation: https://github.com/ocaml/ocaml/pull/10409#issuecomment-847118254. Most notably, stack overflow detection probably has to remain disabled on iOS.

**Comment by xavierleroy** [comment]
Thanks for the report.  This is fixed in trunk.  We'll see if the fix can be released as part of version 4.13.
